src/classes/HiFiMixerSession.ts

```diff
@@ -370,13 +370,13 @@ export class HiFiMixerSession {
                 });
 
                 let mixerPeerKeys = Object.keys(this._mixerPeerKeyToStateCacheDict);
-                for (const mixerPeerKey of mixerPeerKeys) {
-                    if (this._mixerPeerKeyToStateCacheDict[mixerPeerKey].hashedVisitID === hashedVisitID) {
-                        if (this._mixerPeerKeyToStateCacheDict[mixerPeerKey].providedUserID) {
-                            deletedUserData.providedUserID = this._mixerPeerKeyToStateCacheDict[mixerPeerKey].providedUserID;
+                for (const key of mixerPeerKeys) {
+                    let peerData = this._mixerPeerKeyToStateCacheDict[key];
+                    if (peerData.hashedVisitID === hashedVisitID) {
+                        if (peerData.providedUserID) {
+                            deletedUserData.providedUserID = peerData.providedUserID;
                         }
-                        // TODO: remove the entry from the peer state cache -- is this OK?
-                        //delete this._mixerPeerKeyToStateCacheDict[mixerPeerKey];
+                        delete this._mixerPeerKeyToStateCacheDict[key];
```

Comment:
While looking over the code I noticed this **for** loop was performing double map lookups in **_mixerPeerKeyToStateCacheDict** in the common case and triple lookups in the uncommon.  This offends my sensibilities for high performance code and so I was compelled to fix it.

Reply:
Oh yeah, also notice that I did uncomment the code for the old **TODO**.  Yes, it is a good idea to actually purge old peer data from the cache when it gets deleted.